### PR TITLE
[Consignments] Add ArtistConsignButton to Artist page

### PR DESCRIPTION
**Pull request by damassi**
Adds a new ConsignButton to the Artist page, follow up on [recent](https://github.com/artsy/reaction/pull/3329) [work](https://github.com/artsy/reaction/pull/3317) in reaction. 

artist

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Missing Test Files:

- `src/lib/Components/Artist/__tests__/ArtistConsignButton-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "#skip_new_tests".</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Components/Artist/ArtistConsignButton.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { Schema } from "lib/utils/track"
+import { TouchableWithoutFeedback } from "react-native"
+
+export interface ArtistConsignButtonProps {
+  artist: ArtistConsignButton_artist
+  context: React.Component<any>
+}
+
+export const ArtistConsignButton: React.FC<ArtistConsignButtonProps> = props => {
+  const { isInMicrofunnel, imageURL, headline } = getData(props)
```
Comment (ashfurrow) [inline src/lib/Components/Artist/ArtistConsignButton.tsx:19]: Any reason to put this into `getData` instead of inlining that logic in `render()`?

**Comment by ds300** [comment]
:ceiling-cat:

I have some design feedback. This widget immediately jumped out at me as not fitting with the rest of the app.

image

It's all tight and jumbled on the left there, needs more breathing room.

Also not sure about the copy "Sell your {Artist Name}", seems a little awkward to me, but then again I'm not an art collector.

**Comment by ashfurrow** [comment]
Good catch David. The design doesn’t quite match the balance we have in, for example, the EntityHeader component. Maybe we could float the “get started” button to the right?

**Comment by ds300** [comment]
yeah and also the border seems unnecessary in this context

File: src/lib/Components/Artist/ArtistConsignButton.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  const tracking = useTracking()
+
+  return (
+    <TouchableWithoutFeedback
```
Comment (ds300) [inline src/lib/Components/Artist/ArtistConsignButton.tsx:23]: We should be using `TouchableOpacity` here. `TouchableWithoutFeedback` should be used rarely in situations where the user doesn't need extra feedback about whether or not they might trigger an action on press.

File: src/lib/Components/Artist/ArtistConsignButton.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+              </Button>
+            </Box>
+          </Flex>
+        </Flex>
+      </BorderBox>
+    </TouchableWithoutFeedback>
+  )
+}
+
+function getData(props) {
```
Comment (ds300) [inline src/lib/Components/Artist/ArtistConsignButton.tsx:66]: we're going to be enabling `noImplicitAny` soon so would be good to make sure all functions have appropriate parameter types.

File: data/schema.graphql [hunk 4]
```diff
@@ -5407,89 +5375,42 @@ type Mutation {
   ): RequestCredentialsForAssetUploadPayload
 
   # Attach an gemini asset to a consignment submission
-  createGeminiEntryForAsset(
-    input: CreateGeminiEntryForAssetInput!
-  ): CreateGeminiEntryForAssetPayload
+  createGeminiEntryForAsset(input: CreateGeminiEntryForAssetInput!): CreateGeminiEntryForAssetPayload
```
Comment (ds300) [inline data/schema.graphql:5379]: all these formatting changes strike me as odd. any idea what's going on here?

File: CHANGELOG.yml [hunk 5]
```diff
@@ -4,6 +4,7 @@ upcoming:
   dev:
     - Added shell for Galleries development of Viewing Rooms - ash
     - Fixes Clear All functionality in collections filtering - sweir27
+    - Adds <ArtistConsignButton> to Artist page - chris
```
Comment (ashfurrow) [inline CHANGELOG.yml:7]: We'll meed to move this up to `upcoming` to merge in changelog changes in #3109.

File: src/lib/Components/Artist/ArtistConsignButton.tsx (hunk 1 above)
Comment (jonallured) [reply src/lib/Components/Artist/ArtistConsignButton.tsx:19]: FWIW I like that we have extracted some logic to `getData` as has better encapsulation but totally defer to @ashfurrow in terms of following the conventions of Eigen. Small functions like this are also really easy to get specs around!

Comment (ds300) [reply src/lib/Components/Artist/ArtistConsignButton.tsx:19]: I'd vote for inlining. `getData` is hardly doing anything so having it separated out actually makes it harder to follow the logic of this render function IMO.

Comment (damassi) [reply src/lib/Components/Artist/ArtistConsignButton.tsx:19]: Ah, this was copy/paste from reaction. Will fix.

**Comment by damassi** [comment]
@ds300 - these are the designs. They're trying to surface consignments more. Feel free to raise in #consignments channel however and we can iterate.

**Comment by ds300** [comment]
@damassi I'm offline for the evening but I've asked Sam to talk to Nicole B about it. This design work great on the web but on mobile it's awkward and not QWOA so let's iterate before merging :+1:

File: src/lib/Components/Artist/ArtistConsignButton.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+import { ArtistConsignButton_artist } from "__generated__/ArtistConsignButton_artist.graphql"
+import SwitchBoard from "lib/NativeModules/SwitchBoard"
+import { Router } from "lib/utils/router"
+import { Schema } from "lib/utils/track"
+import { TouchableWithoutFeedback } from "react-native"
+
+export interface ArtistConsignButtonProps {
+  artist: ArtistConsignButton_artist
+  context: React.Component<any>
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/lib/Components/Artist/ArtistConsignButton.tsx:15]: Cool, that worked good. Was a bit confusing in the sense that the `ref` needed to be assigned to the container but in retrospect that makes sense.

File: data/schema.graphql (hunk 4 above)
Comment (damassi) [reply data/schema.graphql:5379]: Prettier? Not too sure.

**Comment by damassi** [comment]
@ds300 - as discussed in [this thread](https://artsy.slack.com/archives/CS3H1K4BD/p1585768533202600) we're going to merge and iterate. Thank you for raising as well as the detailed review.

**Pull request merged**